webapp/javascript/components/Toolbar.tsx

```diff
@@ -179,22 +193,88 @@ function FocusOnSubtree({
   );
 }
 
-function HighlightSearch({ onHighlightChange, showMode }) {
+function HighlightSearch({
+  onHighlightChange,
+  showMode,
+  viewSide,
+  isSearchLinked,
+  linkedSearchQuery,
+  toggleLinkedSearch,
+  resetLinkedSearchSide,
+}) {
+  const [text, setText] = useState('');
+  const searchInputType =
+    isSearchLinked === undefined ? 'standalone' : 'linked';
+
+  if (isSearchLinked === true) {
+    if (viewSide === resetLinkedSearchSide) {
+      if (text !== '') {
+        setText('');
+      }
+    } else if (text !== linkedSearchQuery) {
+      setText(linkedSearchQuery);
+    }
+  }
+
+  const testId = `flamegraph-search${viewSide ? `-${viewSide}` : ''}`;
+
+  const searchSizeCls = `${
+    showMode === 'small' ? `${styles['search-small']}` : ''
+  }`;
+
+  const inputTypeCls = `${
+    searchInputType === 'linked' ? `${styles['linked-search-input']}` : ''
+  }`;
+
+  const linkedSearchCls = `${
+    isSearchLinked === true ? `${styles.active}` : ''
+  }`;
+
   return (
-    <DebounceInput
-      data-testid="flamegraph-search"
-      className={`${styles.search} ${
-        showMode === 'small' ? styles['search-small'] : ''
-      }`}
-      type="search"
-      name="flamegraph-search"
-      placeholder="Search…"
-      minLength={2}
-      debounceTimeout={100}
-      onChange={(e) => {
-        onHighlightChange(e.target.value);
-      }}
-    />
+    <>
+      <DebounceInput
+        data-testid={`${testId}`}
+        className={`${styles.search} ${searchSizeCls} ${inputTypeCls} ${linkedSearchCls}`}
+        type="search"
+        name="flamegraph-search"
+        placeholder="Search…"
+        minLength={2}
+        value={isSearchLinked === true ? linkedSearchQuery : text}
+        debounceTimeout={100}
+        onChange={(e) => {
+          onHighlightChange(e.target.value);
+        }}
+      />
+      {searchInputType === 'linked' && (
+        <button
+          data-testid={`link-search-btn-${viewSide}`}
+          type="button"
+          title={`${
+            isSearchLinked === true ? 'Unsync search term' : 'Sync search term'
+          }`}
```

Comment:
Not a big fan of using titles for this cases, but I guess until we have proper tooltip support this is fine.

Reply:
agree. we can replace title once we have a generic tooltip.